### DISCUSSION TITLE
App AuxKernel value changed with new MOOSE build

### DISCUSSION THREAD

**Discussion by smpark7** [idaholab/moose#28434] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I am currently working on bringing the Moltres app up to date with the latest MOOSE [build](https://github.com/idaholab/moose/commit/bc8f029412f6ff91c0973a47ddca420d3928baa0) and moose-dev=2024.08.12.

We have tests containing an AuxKernel (`WallShearStressAux`) that we developed. With an old MOOSE build from March 6 2024, the auxkernel along a boundary evaluates to values ranging from 0.0007 to 0.016, which matches our manual calculations. With the latest MOOSE build, the auxkernel evaluates to values ranging from 0.008 to 5.4. As far as I can tell, all other variables and values match our gold copy.

Could there have been any changes to MOOSE that affected the auxkernel? E.g., monomial constant auxvariable value computations along boundaries, retrieval of variable gradients for auxkernel computations, etc.

Our source code for WallShearStressAux is [here](https://github.com/arfc/moltres/blob/devel/src/auxkernels/WallShearStressAux.C).  It works with vector velocity variables from the INSAD system.

**Comment by GiudGiud** [comment]
Hello

Where are the diffs happening? In a regular external boundary? Or an internal boundary? Or an interface?

@lindsayad

**Reply by smpark7** [reply]
The WallShearStressAux is limited to the `top` boundary with `boundary = top`, which is a regular external wall boundary.

**Reply by lindsayad** [reply]
I cannot think of anything off the top of my head. This test case must not have been checked in? Otherwise we would have caught the change in CI

**Reply by smpark7** [reply]
Could it be because the auxkernel is only in our heavy tests?

Probably unrelated, but we recently accidentally broke the CIVET automated merge to master for Moltres because we wanted to limit push privileges of our group users. Do you know how to fix this without lifting the push restrictions?

In the meantime, I'll try building Moltres with various MOOSE builds to pinpoint the original issue.

**Reply by lindsayad** [reply]
If it's only tested with heavy, then yea that would be why we missed it when testing MOOSE. I will have to ask @milljm or @loganharbour about the push issue

**Reply by smpark7** [reply]
After some testing, we found that the tests pass when compiled with [MOOSE from July 24](https://github.com/idaholab/moose/tree/9a7e0e1d7c085296a28c0506219b649d7779dd75) and moose-dev=2024.07.19. But tests start failing with moose-dev=2024.07.27 (moose version is unchanged). The only significant change between the two moose-dev packages appears to be libmesh.

[These CIVET test results](https://civet.inl.gov/job/2418724/) show the exodiff errors encountered. (Note these tests were on a different MOOSE version but the exodiff error is exactly the same.)

Relevant info about WallShearStressAux in the tests:
It is an auxkernel defined on the top (external) boundary for a constant monomial auxvariable. It computes the magnitude of wall shear stress using gradient values from vector velocity variables (FE) and the boundary normal vector.

**Reply by lindsayad** [reply]
great thanks @smpark7 . This should be enough information for me to fairly easily bisect libmesh

**Reply by smpark7** [reply]
Thanks for investigating this! Another detail that may be of interest: These are the only tests that use CartesianMeshGenerator.

**Reply by lindsayad** [reply]
Will really need to get non-heavy versions of these tests into moltres in the future both for CI coverage (so we know exactly when a breakage occurred in MOOSE) and for ease of doing thinks like bisecting. Have to wait a long time for the test result

**Reply by lindsayad** [reply]
I checked out the MOOSE hash you pointed too and also checked out the libmesh update that occurred when we changed to a moose-dev version of 7.27.24 in https://github.com/idaholab/moose/pull/28244. Your exodiffing tests pass for me there

**Reply by lindsayad** [reply]
After running my own bisect, it appears to me that the "bad commit" is the merge of @GiudGiud's #28115

**Reply by lindsayad** [reply]
Sigh, well I guess you did not make clear what particular tests you were referring to in your original post. #28115 is responsbile for the `pre` heavy tests that are failing, and it was one of those that I based my bisect on. I can see that #28115 is not the culprit for the `sa-model` tests, which are presumably the tests you were referring to in your OP

**Reply by lindsayad** [reply, marked as answer]
For the `sa-model` tests, https://github.com/arfc/moltres/pull/284 will fix it

**Reply by smpark7** [reply]
Sorry I wasn't aware of that distinction between normal and heavy tests. I'll work on the normal test coverage for `pre` and `sa-model` asap and regold `pre` heavy tests for #28115

**Reply by GiudGiud** [reply]
https://github.com/idaholab/moose/pull/28115 is a bug fix btw. So a regold is appropriate.
If for some reason (for example you were trying to use the multiapps for pseudo transients) you do not want to regold, then you can use the no_restore multiapp parameter to get the same results as before

**Reply by smpark7** [reply]
Thanks for the tip! Does #28115 affect simulation results of multiapp transients with tight convergence criteria? I get the impression that it only affects the initial conditions at the start of picard iterations.

**Reply by GiudGiud** [reply]
yes it affects results for all transient multiapps that did not use sub-cycling and have stateful properties or any kind of results obtained on timestep_begin
